packages/rest/test/routes.test.ts

```diff
@@ -86,6 +86,123 @@ describe("rest: routes", () => {
+    const routes = await getRoutesFor(
+      `
+      @route("/things")
+      namespace Things {
+        @get op GetThing(): string;
+
+        @routeReset("/createThing/{thingId}")
+        @put op CreateThing(@path thingId: string): string;
+      }
+      `
+    );
+
+    deepStrictEqual(routes, [
+      { verb: "get", path: "/things", params: [] },
+      { verb: "put", path: "/createThing/{thingId}", params: ["thingId"] },
+    ]);
+  });
+
+  it("routeReset enables a container to reset route prefix for all contained operations", async () => {
+    const routes = await getRoutesFor(
+      `
+      @route("/things")
+      namespace Things {
+        @get op GetThing(): string;
+
+        @route("/{thingId}")
+        @put op CreateThing(@path thingId: string): string;
+
+        @routeReset("/newRoot/{thingId}/subthings")
+        interface Subthing {
+          @get GetSubthing(@path thingId: string): string;
+
+          @route("/{subthingId}")
+          @post CreateSubthing(@path thingId: string, @path subthingId: string): string;
+        }
+      }
+      `
+    );
+
+    deepStrictEqual(routes, [
+      { verb: "get", path: "/things", params: [] },
+      { verb: "put", path: "/things/{thingId}", params: ["thingId"] },
+      { verb: "get", path: "/newRoot/{thingId}/subthings", params: ["thingId"] },
+      {
+        verb: "post",
+        path: "/newRoot/{thingId}/subthings/{subthingId}",
+        params: ["thingId", "subthingId"],
+      },
+    ]);
+  });
+
+  it("routeReset can be used on an autoRoute container", async () => {
+    const routes = await getRoutesFor(
+      `
+      @route("/things")
+      namespace Things {
+        @autoRoute
+        @routeReset("/newRoot")
+        interface Subthing {
+          @segment("subthings")
+          @get GetSubthing(
+            @segment("things")
+            @path thingId: string
+          ): string;
+
+          @post CreateSubthing(
+            @segment("things")
+            @path thingId: string,
+
+            @segment("subthings")
+            @path subthingId: string
+          ): string;
+        }
+      }
+      `
+    );
+
+    deepStrictEqual(routes, [
+      { verb: "get", path: "/newRoot/things/{thingId}/subthings", params: ["thingId"] },
+      {
+        verb: "post",
+        path: "/newRoot/things/{thingId}/subthings/{subthingId}",
+        params: ["thingId", "subthingId"],
+      },
+    ]);
+  });
+
+  it("route and routeReset decorators can override autoRoute on operations", async () => {
+    const routes = await getRoutesFor(
+      `
+      @route("/service")
+      namespace Things {
+        @autoRoute
+        @post op CreateThingBase<TResponse>(
+          @segment("things")
+          @path thingId: string
+        ): TResponse;
+
+        op CreateThing is CreateThingBase<{}>;
+
+        @route("/nonReset/{thingId}")
+        op NotResetCreateThing is CreateThingBase<{}>;
+
+        @routeReset("/reset/{thingId}")
```

Comment:
I feel like I don't like `@routeReset` feels to stateful. Could this be `@absoluteRoute` as this is basically what it does right?

I also think this is a bad feature. Make the interface not portable. Feels like the kind of thing that we'll want a linter rule to discourage the use of it.

Reply:
I meant the name @routeReset kind of make think that there is some state somewhere getting reset.

Reply:
Oh!  I misunderstood.  `@absoluteRoute` could be better, there's also `@routeOverride`

Reply:
For the design discussion about this I think the way you should really design this is like that

```cadl
interface MyOperationGroup extends Path1OPs, Path2Ops {}


namespace Private {
   @route("/path1")
   interface Path1Ops {
      a(): string;
   }
   
   @route("/path1")
   interface Path2Ops {
     a(): string;
  }
}

```